clients/unshaded/src/main/java/tachyon/client/file/FileOutStream.java

```diff
@@ -136,6 +137,22 @@ public void close() throws IOException {
       }
     }
 
+    canComplete = canComplete || storeToTachyon();
+
+    if (canComplete) {
```

Comment:
what if `!canComplete` ? Don't we need to handle that? It looks to me that currently we directly set `mClosed=true`